### ISSUE TITLE
MySQL 8.0.22: Cannot convert value of type `boolean` to integer

### ISSUE THREAD

**Issue by hagen00** [cakephp/cakephp#15185]
This is a:

* [x] bug
* [ ] enhancement
* [ ] feature-discussion (RFC)

* CakePHP Version: 3.9.4
* MySQL 8.0.22

### The Problem
* The table `categories` contains a field `active`, tinyint(1)
* We have a finder `$this->Categories->findByActive(true)`
* This worked fine, when using MySQL 5.7, but now fails with `Cannot convert value of type `boolean` to integer`
* Using MySQL 8.0.22, the following works fine: `$this->Categories->findByActive(1)`

### What you expected to happen
I expect tinyint still to be treated as bool, so that `findByActive(true)` still works. 

The problem may be related to:
[MySQL 8.0.19 release notes](https://dev.mysql.com/doc/relnotes/mysql/8.0/en/news-8-0-19.html)
```
Display width specification for integer data types was deprecated in MySQL 8.0.17
```

**Comment by ADmad** [comment]
> Display width specification for integer data types was deprecated in MySQL 8.0.17

The same point later mentions that TINYINT (1) is exempted from this deprecation.

> MySQL Connectors make the assumption that TINYINT(1) columns originated as BOOLEAN columns; this exception enables them to continue to make that assumption.

**Comment by hagen00** [comment]
Thanks for the feedback. I can confirm that by changing the column from tinyint  to boolean, it now works. 

We used to setup our booleans in MySQL as tinyint (1). Which now no longer works. The MySQL 5.7 schema, that I imported into MySQL8, just treats those as tinyint(0) columns and the code specifically treats those as TYPE_TINYINTEGER and not TYPE_BOOLEAN.
```
// MysqlSchema.php
if (($col === 'tinyint' && $length === 1) || $col === 'boolean') {
    return ['type' => TableSchema::TYPE_BOOLEAN, 'length' => null]
}
```
I suppose this is a mistake on our side and and we should have preferred `BOOLEAN` over `TINYINT(1). The fix will be for us to convert all our tinyint to boolean types, which is more correct. 

I will close this. If anything, this could be noted down in the docs.
